controllers/spacebindingcleanup/spacebinding_cleanup_controller_test.go

```diff
@@ -25,22 +26,70 @@ import (
 
 	"github.com/stretchr/testify/require"
 	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+type publicViewerConfigMock struct{ enabled bool }
+
+func (m publicViewerConfigMock) Enabled() bool {
+	return m.enabled
+}
+
+func newPublicViewerConfigProvider(enabled bool) PublicViewerConfigProvider {
+	return func(context.Context, client.Client) PublicViewerConfig {
+		return publicViewerConfigMock{enabled: enabled}
+	}
+}
+
+var (
+	publicViewerConfigEnabledMockProvider  PublicViewerConfigProvider = newPublicViewerConfigProvider(true)
+	publicViewerConfigDisabledMockProvider PublicViewerConfigProvider = newPublicViewerConfigProvider(false)
+)
+
```

Comment:
As mentioned in the other comment we could avoid those mock functions and just construct the `toolchainconfig` object as required in the specific tests, something like:

```
toolchainconfig := commonconfig.NewToolchainConfigObjWithReset(t,
		testconfig.PublicViewerConfig().Enabled(true))
```
and pass this toolchaincofig to the host fake client. 
This looks more inline with what we currently do in the existing tests.

Reply:
that makes sense, let me fix that

Reply:
done in 022a04807f71b648f7564ca41bc0e9f4e16b9d29